spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/registry/DataFlowUriRegistryPopulator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.registry;
+
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.cloud.deployer.resource.registry.UriRegistry;
+import org.springframework.cloud.deployer.resource.registry.UriRegistryPopulator;
+import org.springframework.util.Assert;
+
+/**
+ * A {@link UriRegistryPopulator} implementation that provides stream and task app URIs for a Data Flow Server.
+ *
+ * @author Mark Fisher
+ */
+public class DataFlowUriRegistryPopulator extends UriRegistryPopulator implements InitializingBean {
```

Comment:
Instead of extending `UriRegistryPopulator` how about passing an instance in as a bean in the constructor?